### PR TITLE
HTCONDOR-1870 add job nature docs

### PR DESCRIPTION
**Pull request by GregThain**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: docs/users-manual/quick-start-guide.rst [hunk 1]
```diff
@@ -28,6 +28,25 @@ This guide presumes that
   (seen on the display), and standard error (seen on the display) may still
   be used, but their contents will be redirected from/to files.
 
+What is a Job?
+--------------
+
+Job is a very specific term in HTCondor. A job is the atomic unit of work.
+A job may use multiple cores on a machine, but one job may not (in general)
+run across more than one machine.  To effectively user HTCondor, you will
```
Comment (Todd-L-Miller) [inline docs/users-manual/quick-start-guide.rst:36]:
```suggestion
run across more than one machine.  To effectively use HTCondor, you will
```

File: docs/users-manual/quick-start-guide.rst [hunk 2]
```diff
@@ -28,6 +28,25 @@ This guide presumes that
   (seen on the display), and standard error (seen on the display) may still
   be used, but their contents will be redirected from/to files.
 
+What is a Job?
+--------------
+
+Job is a very specific term in HTCondor. A job is the atomic unit of work.
+A job may use multiple cores on a machine, but one job may not (in general)
```
Comment (Todd-L-Miller) [inline docs/users-manual/quick-start-guide.rst:35]:
```suggestion
A job may use multiple cores on one machine, but one job may not (in general)
```

File: docs/users-manual/quick-start-guide.rst (hunk 1 above)
Comment (Todd-L-Miller) [inline docs/users-manual/quick-start-guide.rst:36]:
```suggestion
use more than one machine at the same time.  To effectively user HTCondor, you will
```

File: docs/users-manual/quick-start-guide.rst [hunk 3]
```diff
@@ -28,6 +28,25 @@ This guide presumes that
   (seen on the display), and standard error (seen on the display) may still
   be used, but their contents will be redirected from/to files.
 
+What is a Job?
+--------------
+
+Job is a very specific term in HTCondor. A job is the atomic unit of work.
```
Comment (Todd-L-Miller) [inline docs/users-manual/quick-start-guide.rst:34]:
```suggestion
"Job" is a very specific term in HTCondor. A job is the atomic unit of work.
```

File: docs/users-manual/quick-start-guide.rst (hunk 1 above)
Comment (Todd-L-Miller) [reply docs/users-manual/quick-start-guide.rst:36]: This is the only "required" fix. ;)

File: docs/users-manual/quick-start-guide.rst [hunk 4]
```diff
@@ -28,6 +28,25 @@ This guide presumes that
...
+A job may use multiple cores on a machine, but one job may not (in general)
+run across more than one machine.  To effectively user HTCondor, you will
+need to divide your total work (often called a workflow) into a number
+of jobs, which can run asynchronously with respect to each other, but which
+may be connected by input and output files.  Each job is described by a
```
Comment (Todd-L-Miller) [inline docs/users-manual/quick-start-guide.rst:39]:
```suggestion
of jobs.  These atomic units of work run asynchronously with respect to each other, but 
may be connected by input and output files.  Each job is described by a
```

File: docs/users-manual/quick-start-guide.rst [hunk 5]
```diff
@@ -28,6 +28,25 @@ This guide presumes that
...
+Job ClassAd, which is usually created by the system from a submit description file.
```
Comment (Todd-L-Miller) [inline docs/users-manual/quick-start-guide.rst:40]: Why is 'Job' capitalized here?

File: docs/users-manual/quick-start-guide.rst [hunk 6]
```diff
@@ -28,6 +28,25 @@ This guide presumes that
...
+HTCondor is a High Throughput system, which means it has been designed to 
+effectively manage hundreds of thousands of jobs.  Attributes of jobs that
+must be defined include the executable or script to run, the amount of memory, CPU
+and other machine resources it needs, and descriptions of the file inputs it need.
+The set of files used by a job is called the "sandbox".  There is an input sandbox,
+the input files that exist before a job starts; the output sandbox, the set of files
+created by the job; and a scratch sandbox, the set of files made as the job runs.
```
Comment (Todd-L-Miller) [inline docs/users-manual/quick-start-guide.rst:47]: Don't we normally call the 'scratch sandbox' the 'job sandbox'?

**Comment by Todd-L-Miller** [comment]
I have issues with the first few paragraphs of this documentation, which I may attempt to resolve on this PR for our mutual convenience.

File: docs/users-manual/quick-start-guide.rst (hunk 6 above)
Comment (GregThain) [reply docs/users-manual/quick-start-guide.rst:47]: Feel like "job sandbox" is a bit confusing wrt the various places a sandbox can live.

File: docs/users-manual/quick-start-guide.rst (hunk 5 above)
Comment (GregThain) [reply docs/users-manual/quick-start-guide.rst:40]: I think my editor did that...

**Review by Todd-L-Miller — APPROVED** [review]

**Pull request merged**